common/djangoapps/student/tests/tests.py

```diff
@@ -930,3 +933,182 @@ def test_roundtrip_with_unicode_course_id(self):
+                'status': course[1],
+                'course_codes': [
+                    {
+                        'display_name': 'Demo XSeries Program 1',
+                        'key': 'TEST_A',
+                        'run_modes': [{'sku': '', 'mode_slug': 'ABC_1', 'course_key': 'edX/DemoX_1/Run_1'}]
+                    }
+                ],
+                'subtitle': 'Dummy program 1 for testing',
+                'name': 'First Program'
+            }
+
+        return programs
+
+    @ddt.data(
+        ('active', [{'sku': ''}, {'sku': ''}, {'sku': ''}, {'sku': ''}], 4, 'slug1'),
+        ('active', [{'sku': ''}, {'sku': ''}, {'sku': ''}], 3, 'slug2'),
+        ('active', [], 0, ''),
+        ('inactive', [{'sku': ''}, {'sku': ''}, {'sku': ''}, {'sku': ''}], 4, 'slug3'),
+    )
+    @ddt.unpack
+    def test_get_xseries_programs_method(self, status, course_codes, course_codes_count, slug):
+        with patch('student.views._get_course_programs') as mock_data:
+            mock_data.return_value = {
+                unicode('course-v1:ManTestX+ManTest2+2014'): {
+                    'category': 'xseries',
+                    'organization': {'display_name': 'Test Organization 1', 'key': 'edX'},
+                    'marketing_slug': slug,
+                    'status': status,
+                    'course_codes': course_codes,
+                    'subtitle': 'Dummy program 1 for testing',
+                    'name': 'First Program'
+                }
+            }
+            parse_data = _get_course_programs(
+                self.user, [
+                    u'course-v1:ManTestX+ManTest2+2014', 'edx/demox/DemoCourse'
+                ]
+            )
+
+            if status == 'inactive':
+                self.assertDictEqual({}, parse_data)
+            else:
+                self.assertDictEqual(
+                    {
+                        'course-v1:ManTestX+ManTest2+2014': {
+                            'course_count': course_codes_count,
+                            'display_name': 'First Program',
+                            'program_marketing_url': urljoin(
+                             settings.MKTG_URLS.get('ROOT'), 'xseries' + '/{}').format(slug)
+                        }
+                    },
+                    parse_data
+                )
+
+    @unittest.skipUnless(settings.ROOT_URLCONF == 'lms.urls', 'Test only valid in lms')
+    def test_program_courses_on_dashboard_without_configuration(self):
+        """If programs configuration is disabled than xseries upsell messages
+        will not appear on student dashboard.
+        """
+        CourseEnrollment.enroll(self.user, self.course.id)
+        self.client.login(username="jack", password="test")
+        response = self.client.get(reverse('dashboard'))
+        self.assertEquals(response.status_code, 200)
+        self.assertIn('Pursue a Certificate of Achievement to highlight', response.content)
+        self._assert_responses(response, 0)
```

Comment:
for this test, please add a mock on `get_course_programs_for_dashboard` and then assert that it was NOT called - this gives us confidence that our feature switch in the dashboard view is working correctly.